magnet_coils.py

```diff
@@ -0,0 +1,350 @@
+            definition. Note that the cross-section shape must be either a
+            circle or rectangle.
+            For a circular cross-section, the list format is
+            ['circle' (str), radius (float, cm)]
+            For a rectangular cross-section, the list format is
+            ['rectangle' (str), width (float, cm), thickness (float, cm)]
+
+    Returns:
+        vol_id (list of int): indices for magnet volumes.
+    """
+    # Extract coil cross-section shape
+    shape = cross_section[0]
+    # Cross-section inititally populated with thickness vector
+    # oriented along x axis
+    t_vec = np.array([1, 0, 0])
+    # Conditionally extract parameters for circular cross-section
+    if shape == 'circle':
+        if len(cross_section) != 2:
+            logger.debug(
+                'Format of list defining circular cross-section must be\n'
+                '[\'circle\' (str), radius (float, cm)]'
+            )
+            raise ValueError(
+                'Format of list defining circular cross-section must be\n'
+                '[\'circle\' (str), radius (float, cm)]'
+            )
+        radius = cross_section[1]
+    # Conditinally extract parameters for rectangular cross-section
+    elif shape == 'rectangle':
+        if len(cross_section) != 3:
+            logger.debug(
+                'Format of list defining rectangular cross-section must be\n'
+                '[\'rectangle\' (str), width (float, cm), thickness '
+                '(float, cm)]'
+            )
+            raise ValueError(
+                'Format of list defining rectangular cross-section must be\n'
+                '[\'rectangle\' (str), width (float, cm), thickness '
+                '(float, cm)]'
+            )
+        width = cross_section[1]
+        thickness = cross_section[2]
+    else:
+        logger.debug(
+            'Magnet cross-section must be either a circle or rectangle. The '
+            'first entry of the list defining the cross-section must be the '
+            'shape, with the following entries defining the shape parameters.\n'
+            '\n'
+            'For a circular cross-section, the list format is\n'
+            '[\'circle\' (str), radius (float, cm)]\n'
+            '\n'
+            'For a rectangular cross-section, the list format is\n'
+            '[\'rectangle\' (str), width (float, cm), thickness (float, cm)]'
+        )
+        raise ValueError(
+            'Magnet cross-section must be either a circle or rectangle. The '
+            'first entry of the list defining the cross-section must be the '
+            'shape, with the following entries defining the shape parameters.\n'
+            '\n'
+            'For a circular cross-section, the list format is\n'
+            '[\'circle\' (str), radius (float, cm)]\n'
+            '\n'
+            'For a rectangular cross-section, the list format is\n'
+            '[\'rectangle\' (str), width (float, cm), thickness (float, cm)]'
+        )
```

Comment:
This might make a good function, especially if it just returned a string, `shape_str`, that was either:
* `f'{shape} radius {radius}'` or 
* `f'{shape} width {thickness} height {width}'`

so you could then just call

`cubit.cmd('create surface ' + shape_str + ' zplane')`


(Note: `t_vec` should be defined outside of this function, either before or after...)